parallel build with gnu make 4.4 fails

I seem to get a build issue when using GNU make 4.4 with

```
'make' 'MAKE_NB_JOBS=0' 'ARCH=x86_64' 'TARGET=ZEN' 'USE_LOCKING=1' 'USE_OPENMP=1' 'USE_THREAD=1' 'libs' 'netlib' 'shared'
```

with `-j16` at the top-level make (openblas is built with a recursive make; not sure if relevant) which fails after some time with

```
     29      Warning: Possible change of value in conversion from REAL(4) to INTEGER(4) at (1) [-Wconversion]
     30      sgebak.f:253:19:
     31      
     32        253 |                K = SCALE( I )
     33            |                   1
     34      Warning: Possible change of value in conversion from REAL(4) to INTEGER(4) at (1) [-Wconversion]
  >> 35      make[2]: *** No rule to make target '../libopenblas_zenp-r0.3.21.a', needed by '../libopenblas_zenp-r0.3.21.so'.  Stop.
     36      make[2]: Leaving directory '/tmp/user/spack-stage/spack-stage-openblas-0.3.21-rqvfhzdue2wzonslpabua3qssw7t6mu2/spack-src/exports'
  >> 37      make[1]: *** [Makefile:125: shared] Error 2
     38      make[1]: *** Waiting for unfinished jobs....
     39      /home/user/spack/lib/spack/env/gcc/gfortran -O2 -Wall -frecursive -fno-optimize-sibling-calls -m64 -fopenmp -fPIC -msse3 -msse4.1 -mavx -mavx2 -mavx2 -c -o sgebd2.o sgebd2.f
...
```

Is there anything that jumps out of the changelog that could explain this? https://lists.gnu.org/archive/html/info-gnu/2022-10/msg00008.html. 

<details>
<summary>Maybe this?</summary>

>  Previously each target in a explicit grouped target rule was considered
>  individually: if the targets needed by the build were not out of date the
>  recipe was not run even if other targets in the group were out of date.  Now
>  if any of the grouped targets are needed by the build, then if any of the
>  grouped targets are out of date the recipe is run and all targets in the
>  group are considered updated.
</details>

With GNU make 4.3 everything looks fine.

Ah, if I do `make MAKE_NB_JOBS=0 -j16 libs && make MAKE_NB_JOBS=0 -j16 shared` in that order, it errors:

```
home/harmen/spack/lib/spack/env/gcc/gcc -O2 -DSMALL_MATRIX_OPT -DMAX_STACK_ALLOC=2048 -Wall -m64 -DF_INTERFACE_GFORT -fPIC -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=32 -DMAX_PARALLEL_NUMBER=1 -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.21\" -msse3 -mssse3 -msse4.1 -mavx -mavx2 -mavx2 -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME= -DASMFNAME=_ -DNAME=_ -DCNAME= -DCHAR_NAME=\"_\" -DCHAR_CNAME=\"\" -DNO_AFFINITY -I..  -w -o linktest linktest.c ../libopenblas_zenp-r0.3.21.so -L/usr/lib/gcc/x86_64-linux-gnu/12 -L/usr/lib/gcc/x86_64-linux-gnu/12/../../../x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/12/../../../../lib -L/lib/x86_64-linux-gnu -L/lib/../lib -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib -L/usr/lib/gcc/x86_64-linux-gnu/12/../../..  -lgfortran -lm -lquadmath -lm -lc  && echo OK.
/usr/bin/ld: /tmp/cct3hvMp.o: in function `main':
linktest.c:(.text.startup+0x536): undefined reference to `iparam2stage_'
/usr/bin/ld: linktest.c:(.text.startup+0x53d): undefined reference to `ilaenv2stage_'
/usr/bin/ld: linktest.c:(.text.startup+0x591): undefined reference to `spotri_'
/usr/bin/ld: linktest.c:(.text.startup+0x5e5): undefined reference to `dpotri_'
/usr/bin/ld: linktest.c:(.text.startup+0x639): undefined reference to `cpotri_'
```

Even simpler
```
$ make clean 1>/dev/null 2>&1 && make --trace shared MAKE_NB_JOBS=0
/usr/bin/ld: warning: /tmp/ccRR3ifI.o: missing .note.GNU-stack section implies executable stack
/usr/bin/ld: NOTE: This behaviour is deprecated and will be removed in a future version of the linker
Makefile:125: target 'shared' does not exist
make -C exports so
make[1]: Entering directory '/tmp/harmen/spack-stage/spack-stage-openblas-0.3.21-zwgrfwv5q7nn3icvqos6nuosxohpcyfk/spack-src/exports'
make[1]: *** No rule to make target '../libopenblas_zenp-r0.3.21.a', needed by '../libopenblas_zenp-r0.3.21.so'.  Stop.
make[1]: Leaving directory '/tmp/harmen/spack-stage/spack-stage-openblas-0.3.21-zwgrfwv5q7nn3icvqos6nuosxohpcyfk/spack-src/exports'
make: *** [Makefile:125: shared] Error 2
```

This also happens with make 4.3

--- a/Makefile
+++ b/Makefile
@@ -110,7 +110,7 @@ endif
 	@echo "To install the library, you can run \"make PREFIX=/path/to/your/installation install\"."
 	@echo
 
-shared :
+shared : libs netlib $(RELA)
 ifneq ($(NO_SHARED), 1)
 ifeq ($(OSNAME), $(filter $(OSNAME),Linux SunOS Android Haiku))
 	@$(MAKE) -C exports so
@@ -134,7 +134,7 @@ ifeq ($(OSNAME), CYGWIN_NT)
 endif
 endif
 
-tests :
+tests : libs netlib $(RELA) shared
 ifeq ($(NOFORTRAN), $(filter 0,$(NOFORTRAN)))
 	touch $(LIBNAME)
 ifndef NO_FBLAS

